src/lib.rs

```diff
@@ -430,6 +432,311 @@ impl<T> OptionExt<T> for Option<T> {
     }
 }
 
+/// Additions to [`Future`](futures::Future).
+pub trait FutureExt<T, E>: Sized {
+    /// Extend a `Future`'s error with additional context-sensitive information.
+    ///
+    /// ```rust
+    /// # extern crate futures;
+    /// use futures::Future;
+    /// # use futures::IntoFuture;
+    /// use snafu::{Context, FutureExt, Snafu};
+    ///
+    /// #[derive(Debug, Snafu)]
+    /// enum Error {
+    ///     Authenticating { user_name: String, user_id: i32, source: ApiError },
+    /// }
+    ///
+    /// fn example() -> impl Future<Item=i32, Error=Context<ApiError, Authenticating<&'static str, i32>>> {
```

Comment:
```suggestion
    /// fn example() -> impl Future<Item = i32, Error = Context<ApiError, Authenticating<&'static str, i32>>> {
```

Reply:
Ditto this elsewhere. Someday I'll figure out how to run rustfmt on code examples, but for now it's all by hand...